src/vs/workbench/browser/parts/compositeBarActions.ts

```diff
@@ -158,7 +158,15 @@ export class ActivityActionViewItem extends BaseActionViewItem {
 		if (this.label) {
 			if (this.options.icon) {
 				const foreground = this._action.checked ? colors.activeBackgroundColor || colors.activeForegroundColor : colors.inactiveBackgroundColor || colors.inactiveForegroundColor;
-				this.label.style.backgroundColor = foreground ? foreground.toString() : '';
+				const isExtension = this.label.className.indexOf('extensionViewlet-') >= 0;
```

Comment:
Total hack.
The `compositeBarActions` is not aware of the names of the viewlets.
`extensionViewlet` and `remote` should not be referenced from here.
Can we tackle this in some nicer way?

Reply:
I totally see your point and this is the only portion that I struggled with. I needed a way to know if the activity bar item is an extension, because it defines a custom svg and needs to be treated differently. We use the mask trick + bg color to theme extension icons. This doesn’t work for an icon font as it needs a ‘color’ instead. 

If you know if other ways to detect if it’s an extension let me know. I’ll see if there are other non-hacky ways to get around this.

Reply:
I don't think this is any better, but I cleaned this up to be:
```ts
const isExtension = this.activity.cssClass?.indexOf('extensionViewlet') === 0;
```

I'll work with @sandy081 to see if there is a cleaner way to do this when he's back